### ISSUE TITLE
Assessment Results

### ISSUE THREAD

**Issue by ap2322** [kevn7n/mod_0_skills#1]
Kevin, you performed well enough on the assessment to merit a chance to make some changes that can promote you to `technical ready`. Please make the following updates by noon MST Saturday:

- [ ] Try running each line of code from `variables.txt` in your terminal using irb. Correct any errors you come across.
- [ ] Double check the variable assignment syntax of your attributes in your class examples (especially booleans). Do you get errors when you assign any of those variables in irb? Correct your syntax on your class examples to account for the correct data type.

**Git fire:** Thanks for communicating how changes are on separate branches. It appears you also have your skills assessment nested in a folder identical to your repo name which is not the expected file structure.
    - [ ] Comment below the errors you made in your git workflow and how you will prevent those mistakes in the future. 
    - [ ] Extension: Copy over your variables.txt file to your main branch and fix your file structure so at the root of the project, all we see is the `classes` directory and `variables.txt`. Timebox to 1 hr and if unsuccessful with this, comment below your search terms and things you've attempted to correct this error.

**Comment by kevn7n** [comment]
Alice.  I corrected the errors in 'variables.txt' after running irb, the variable assignments are corrected.

Git fire:  I recognized my error later a few hours after the assessment, which was when I slack msg'd Ellen about my two branches.  After creating and pushing my four 'classes' files, I realized I mistakenly created the blank 'variables.txt' file outside of the mod_0_skills dir and already pushed the initial commit.  I deleted the file and the .git dir also, after I was not able to push my new updated 'variables.txt' file from inside the mod_0_skills dir.  When I git init again, I did not notice that the branch reverted to 'master' since I forgot to do 'git branch -M main', and pushed it to master.  I wasted a lot of time at first not seeing the second branch, then more time trying to figure out why i did not have access to push again.  By then I was against the 5:30pm deadline, so submitted it as is.

I deleted variables.txt on MASTER, and tried repeatedly to push it to MAIN, making sure it was branched to MAIN with 'git branch -M main', but got errors.  I also tried  'git push git@github.com:kevn7n/mod_0_skills.git' and 'git push --set-upstream git@github.com:kevn7n/mod_0_skills.git main' and 'git fetch' 'git merge' after some googling.  In the future I will be especially vigilant that the correct branch is specified before my git push, now that I'm aware of them diverging.

I did not want to risk losing this Assessment Results note by tampering with the repo.  But to show I can correctly do the assessment, I redid all the steps and work in less than 20 min (copying in the original 4 files in 'classes' and variables.txt) in a new repo by the same name with a '_2' after =>   https://github.com/kevn7n/mod_0_skills_2   I hope that helps show my proficiency, since I now know and have done all the steps many times.  Thank you.

**Comment by ap2322** [comment]
Thanks for taking the deep dive into _how_ you got yourself into a git fire, _what_ you tried to get out of it, and what you plan to do to _prevent_ it in the future. Really great work communicating on this.

One last thing I want you to check regards booleans, a basic data type:
Which is correct Ruby syntax, a or b?
a.) `is_clean = f`
b.) `is_clean = false`

Please comment below, thanks.

**Comment by kevn7n** [comment]
Thanks Alice for the consideration and thorough review.  b.)  is_clean = false.    Boolean values can be true or false, not shortened to t or f, and always lowercase.  I need to slow down during tests and probably take a minute to reset, for a final review before submitting tests.  Thank you again.

**Comment by ap2322** [comment]
Yes, that's correct. And those are great strategies for tests. You're officially `technical ready`. Here are some things to keep in mind as you move forward:

- Good work following Ruby conventions around naming classes (PascalCase, singular), attributes and methods (snake_case), and making your attribute and method names descriptive & brief.
- Keep conceptualizing of a class as an overall blueprint and the object/instance as a single example of that class.
- Paying close attention to Ruby syntax will help your fluency as you learn more programming patterns.
